src/main/scala/codacy/metrics/ScalaMetrics.scala

```diff
@@ -0,0 +1,56 @@
+package codacy.metrics
+
+import java.nio.file.Path
+
+import codacy.docker.api.metrics.{FileMetrics, MetricsTool}
+import codacy.docker.api.{MetricsConfiguration, Source}
+import com.codacy.api.dtos.Language
+
+import scala.util.{Properties, Try}
+
+object ScalaMetrics extends MetricsTool {
+  override def apply(source: Source.Directory,
+                     language: Option[Language],
+                     files: Option[Set[Source.File]],
+                     options: Map[MetricsConfiguration.Key, MetricsConfiguration.Value]): Try[List[FileMetrics]] = {
+
+    Try {
+      val filesSeq: Set[Source.File] = files.getOrElse(allFilesIn(new java.io.File(source.path).toPath))
+
+      filesSeq.map { srcFile =>
+        val fileWithFullPath = Source.File(source.path + "/" + srcFile.path)
+
+        val (classCount, methodCount) = classesAndMethods(fileWithFullPath) match {
+          case Some((classes, methods)) => (Some(classes), Some(methods))
+          case _                        => (None, None)
+        }
+
+        FileMetrics(filename = srcFile.path, nrClasses = classCount, nrMethods = methodCount)
+      }(collection.breakOut)
+    }
+  }
+
+  private def allFilesIn(srcPath: Path, relativize: Boolean = true): Set[Source.File] = {
```

Comment:
The seed already has this logic, and if not better files has implementation for this.